### PR TITLE
[Merged by Bors] - feat(Analysis/SpecificLimits/Normed): add lemma about geom_series

### PR DESCRIPTION
**Pull request by CBirkbeck**
Add some lemmas about multiplying the geometric series by `x` and `1+x` (where `x` is the term being summed). These are used in #12770 

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/SpecificLimits/Normed.lean [hunk 1]
```diff
@@ -538,6 +538,32 @@ theorem mul_neg_geom_series (x : R) (h : ‖x‖ < 1) : ((1 - x) * ∑' i : ℕ,
...
+theorem geom_series_mul_shift (x : R) (h : ‖x‖ < 1) :
+    x * ∑' i : ℕ, x ^ i = ∑' i : ℕ, x ^ (i + 1) := by
+  refine' tendsto_nhds_unique
+      ((NormedRing.summable_geometric_of_norm_lt_one x h).hasSum.mul_left x).tendsto_sum_nat _
+  have : Tendsto (fun n : ℕ => ∑ i in Finset.range (n + 1), x ^ i - 1) atTop
+      (𝓝 (∑' i : ℕ, x ^ (i + 1))) := by
+    rw [geom_series_succ x h]
+    apply Tendsto.sub_const
+    simp only [sum_range_succ]
+    have hp : tsum (fun (i : ℕ)=> x^i) = tsum (fun (i : ℕ)=> x^i) + 0 := by exact (add_zero _).symm
```
Comment (Ruben-VandeVelde) [inline Mathlib/Analysis/SpecificLimits/Normed.lean:555]:
```suggestion
    have hp : tsum (fun (i : ℕ) => x^i) = tsum (fun (i : ℕ) => x^i) + 0 := (add_zero _).symm
```

**Review by Ruben-VandeVelde — APPROVED** [review]

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

File: Mathlib/Analysis/SpecificLimits/Normed.lean [hunk 2]
```diff
@@ -538,6 +538,32 @@ theorem mul_neg_geom_series (x : R) (h : ‖x‖ < 1) : ((1 - x) * ∑' i : ℕ,
   rw [← mul_neg_geom_sum, Finset.mul_sum]
 #align mul_neg_geom_series mul_neg_geom_series
 
+theorem geom_series_succ (x : R) (h : ‖x‖ < 1) : ∑' i : ℕ, x ^ (i + 1) = ∑' i : ℕ, x ^ i - 1 := by
+  apply symm
+  rw [sub_eq_iff_eq_add, tsum_eq_zero_add (NormedRing.summable_geometric_of_norm_lt_one x h),
+    pow_zero, add_comm]
+
+theorem geom_series_mul_shift (x : R) (h : ‖x‖ < 1) :
+    x * ∑' i : ℕ, x ^ i = ∑' i : ℕ, x ^ (i + 1) := by
```
Comment (MichaelStollBayreuth) [inline Mathlib/Analysis/SpecificLimits/Normed.lean:547]: `h` is not really necessary here (if it doesn't hold, then both sides are zero by definition).

File: Mathlib/Analysis/SpecificLimits/Normed.lean [hunk 3]
```diff
@@ -538,6 +538,32 @@ theorem mul_neg_geom_series (x : R) (h : ‖x‖ < 1) : ((1 - x) * ∑' i : ℕ,
...
+  refine' tendsto_nhds_unique
```
Comment (MichaelStollBayreuth) [inline Mathlib/Analysis/SpecificLimits/Normed.lean:548]: This should be `refine`, I guess.

File: Mathlib/Analysis/SpecificLimits/Normed.lean (hunk 2 above)
Comment (MichaelStollBayreuth) [reply Mathlib/Analysis/SpecificLimits/Normed.lean:547]: (But it appears that the relevant statement wants a division ring, so it's probably not worth removing the condition.)

File: Mathlib/Analysis/SpecificLimits/Normed.lean [hunk 4]
```diff
@@ -538,6 +538,32 @@ theorem mul_neg_geom_series (x : R) (h : ‖x‖ < 1) : ((1 - x) * ∑' i : ℕ,
...
+      (𝓝 (∑' i : ℕ, x ^ (i + 1))) := by
+    rw [geom_series_succ x h]
+    apply Tendsto.sub_const
+    simp only [sum_range_succ]
+    have hp : tsum (fun (i : ℕ) => x^i) = tsum (fun (i : ℕ) => x^i) + 0 := (add_zero _).symm
+    rw [hp]
+    apply Tendsto.add ((NormedRing.summable_geometric_of_norm_lt_one x h).hasSum.tendsto_sum_nat)
+      (tendsto_pow_atTop_nhds_zero_of_norm_lt_one h)
+  convert this
+  rw [@geom_sum_succ _ _ x, mul_sum, add_sub_cancel_right]
```
Comment (MichaelStollBayreuth) [inline Mathlib/Analysis/SpecificLimits/Normed.lean:560]:
```suggestion
  rw [← Summable.tsum_mul_left]
  conv => enter [1, 1, i]; rw [← _root_.pow_succ']
  exact NormedRing.summable_geometric_of_norm_lt_one _ h 
```
(Also, `refine'` is being phased out...)

File: Mathlib/Analysis/SpecificLimits/Normed.lean [hunk 5]
```diff
@@ -538,6 +538,32 @@ theorem mul_neg_geom_series (x : R) (h : ‖x‖ < 1) : ((1 - x) * ∑' i : ℕ,
   rw [← mul_neg_geom_sum, Finset.mul_sum]
 #align mul_neg_geom_series mul_neg_geom_series
 
+theorem geom_series_succ (x : R) (h : ‖x‖ < 1) : ∑' i : ℕ, x ^ (i + 1) = ∑' i : ℕ, x ^ i - 1 := by
+  apply symm
+  rw [sub_eq_iff_eq_add, tsum_eq_zero_add (NormedRing.summable_geometric_of_norm_lt_one x h),
+    pow_zero, add_comm]
```
Comment (MichaelStollBayreuth) [inline Mathlib/Analysis/SpecificLimits/Normed.lean:544]:
```suggestion
  rw [eq_sub_iff_add_eq, tsum_eq_zero_add (NormedRing.summable_geometric_of_norm_lt_one x h),
    pow_zero, add_comm]
```

File: Mathlib/Analysis/SpecificLimits/Normed.lean [hunk 6]
```diff
@@ -538,6 +538,32 @@ theorem mul_neg_geom_series (x : R) (h : ‖x‖ < 1) : ((1 - x) * ∑' i : ℕ,
...
+    rw [hp]
+    apply Tendsto.add ((NormedRing.summable_geometric_of_norm_lt_one x h).hasSum.tendsto_sum_nat)
+      (tendsto_pow_atTop_nhds_zero_of_norm_lt_one h)
+  convert this
+  rw [@geom_sum_succ _ _ x, mul_sum, add_sub_cancel_right]
+
+theorem geom_series_mul_one_add (x : R) (h : ‖x‖ < 1) :
+    (1 + x) * ∑' i : ℕ, x ^ i = 2 * ∑' i : ℕ, x ^ i - 1 := by
+  rw [add_mul, one_mul, geom_series_mul_shift x h, geom_series_succ x h, two_mul]
+  abel
```
Comment (MichaelStollBayreuth) [inline Mathlib/Analysis/SpecificLimits/Normed.lean:565]:
```suggestion
  rw [add_mul, one_mul, geom_series_mul_shift x h, geom_series_succ x h, two_mul, add_sub_assoc]
```
`abel` is overkill here.

File: Mathlib/Analysis/SpecificLimits/Normed.lean (hunk 4 above)
Comment (MichaelStollBayreuth) [reply Mathlib/Analysis/SpecificLimits/Normed.lean:560]: Or in one  line:
```lean
  simp_rw [← (NormedRing.summable_geometric_of_norm_lt_one _ h).tsum_mul_left, ← _root_.pow_succ']
```

Comment (CBirkbeck) [reply Mathlib/Analysis/SpecificLimits/Normed.lean]: Oh that's much nicer, thank you!

**Review by MichaelStollBayreuth — APPROVED** [review]
OK with me now.

**Comment by PatrickMassot** [comment]
bors merge

**Pull request closed without merging**